src/nnbench/reporter/file.py

```diff
@@ -113,33 +113,63 @@ def csv_load(fp: IO, fdoptions: FileDriverOptions) -> list[BenchmarkRecord]:
     # TODO: Add parquet support
 
 
-class FileIO:
-    def __init__(
-        self,
-        drivers: dict[str, SerDe] | None = None,
-        compressions: dict[str, Callable] | None = None,
-    ):
-        super().__init__()
-        self.drivers = drivers or _file_drivers
-        self.compressions = compressions or _compressions
+def get_driver_implementation(name: str) -> SerDe:
+    try:
+        return _file_drivers[name]
+    except KeyError:
+        raise KeyError(f"unsupported file format {name!r}") from None
+
+
+def gzip_compression(filename: str | os.PathLike[str], mode: Literal["r", "w"] = "r") -> IO:
+    import gzip
+
+    # gzip.GzipFile does not inherit from IO[bytes],
+    # but it has all required methods, so we allow it.
+    return cast(IO[bytes], gzip.GzipFile(filename=filename, mode=mode))
+
+
+def bz2_compression(filename: str | os.PathLike[str], mode: Literal["r", "w"] = "r") -> IO:
+    import bz2
+
+    return bz2.BZ2File(filename=filename, mode=mode)
+
 
+with _compression_lock:
+    _compressions["gzip"] = gzip_compression
+    _compressions["bz2"] = bz2_compression
+
+
+def get_compression_algorithm(name: str) -> Callable:
+    try:
+        return _compressions[name]
+    except KeyError:
+        raise KeyError(f"unsupported compression algorithm {name!r}") from None
```

Comment:
Question: Why raise from `None` and not a plain raise?

Reply:
`raise from None` cuts off the parent exception, which is the KeyError in the `_compressions` dict lookup. That does not give you any new information, so I decided to throw it away.